server/src/main/java/org/elasticsearch/index/mapper/RangeFieldMapper.java

```diff
@@ -138,7 +139,13 @@ protected void setupFieldType(BuilderContext context) {
                     Objects.equals(builder.pattern, formatter.pattern()) == false;
 
                 if (hasPatternChanged || Objects.equals(builder.locale, formatter.locale()) == false) {
-                    fieldType().setDateTimeFormatter(DateFormatter.forPattern(pattern).withLocale(locale));
+                    DateFormatter dateTimeFormatter;
+//                    if (Joda.isJodaStyleIndex(context,pattern) ) {
```

Comment:
Shouldn't this be uncommented?

Reply:
yes - I am trying to find a good example where the test for range fields fails in 7 with joda pattern. Luckily the one with 'YYYY' passes -although the values are incorrectly parsed